### PR TITLE
feat: implement relay function

### PR DESCRIPTION
**Pull request by nvtaveras**
This implements the logic for calling the `relay()` function on the relayerAddress parameter from cloudevents. 

I deployed a Mock relayer contract on https://alfajores.celoscan.io/address/0x3005a33a9782f4c1ccfa0ffdb87a034b95b7ad90 and tested that errors are handled gracefully for the following scenarios:

- The contract reverts with a custom defined error
- The contract reverts for any other reason
- The relayer address doesn't have enough balance
- The TX broadcast times out

Fixes: https://github.com/mento-protocol/mento-core/issues/460

### CODE REVIEW

File: infra/cloud_function.tf [hunk 1]
```diff
@@ -24,8 +24,12 @@ resource "google_cloudfunctions2_function" "relay" {
     timeout_seconds       = 60
 
     environment_variables = {
+      GCP_PROJECT_ID       = module.project-factory.project_id
+      RELAYER_PK_SECRET_ID = google_secret_manager_secret.relayer_pk.secret_id
       # Logs execution ID for easier debugging => https://cloud.google.com/functions/docs/monitoring/logging#viewing_runtime_logs
       LOG_EXECUTION_ID = "true"
+      # For testing purposes until we start to deploy on Mainnet
```
Comment (chapati23) [inline infra/cloud_function.tf:31]:
```suggestion
      # TODO: Remove this before go-live
      # For testing purposes until we start to deploy on Mainnet
```

File: src/relay.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  return createWalletClient({
+    account: privateKeyToAccount(pk as Address),
+    chain: isMainnet ? celo : celoAlfajores,
+    transport: http(),
+  });
+}
+
+export default async function relay(relayerAddress: string): Promise<boolean> {
+  const publicClient = getPublicClient();
```
Comment (chapati23) [inline src/relay.ts:38]: given this function is gonna be called pretty often, we should already do some performance optimization here and move the `getPublicClient()` and `getWalletClient()` calls outside the function if possible. Should initialize these clients once and reuse them across function invocations.

https://cloud.google.com/functions/docs/bestpractices/tips#use_global_variables_to_reuse_objects_in_future_invocations

File: src/relay.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      hash,
+      timeout: 50 * 1000, // 10 blocks for the tx to be mined
+    });
+
+    if (receipt.status !== "success") {
+      console.log(`Relay tx failed: ${hash}`);
+      return false;
+    }
+
+    console.log(`Relay tx succeeded: ${hash}`);
```
Comment (chapati23) [inline src/relay.ts:61]: given we might get multiple relay txs in parallel, should we include the rate feed name here for which we relayed?

same comment for line 57 above

**Review by chapati23 — COMMENTED** [review]
great job man, just some questions & suggestions

File: src/relay.ts (hunk 3 above)
Comment (nvtaveras) [reply src/relay.ts:61]: will do this as part of the logging refactoring

File: src/relay.ts (hunk 2 above)
Comment (nvtaveras) [reply src/relay.ts:38]: I'm not sure if we can still do this given that the wallet client initialization is async. Created https://github.com/mento-protocol/oracle-relayer/issues/6 to look into this later

Comment (chapati23) [reply src/relay.ts:38]: yeah for async stuff fair point, need to investigate, left a comment in #6 for a possible approach

**Review by chapati23 — APPROVED** [review]

**Pull request merged**